### ISSUE TITLE
ESP-Matter install is broken (CON-1067)

### ISSUE THREAD

**Issue by S-A-M-J** [espressif/esp-matter#855]
When trying to do a fresh install i get an error in the install.sh script. The problem is also exactly described here: [esp-forum](https://www.esp32.com/viewtopic.php?t=38114) Unfortunately with no answer since end of January. 
I have followed the guide here: [guide](https://docs.espressif.com/projects/esp-matter/en/latest/esp32/developing.html). 
Furthermore i have installed all prerequisites and cloned the repo completely (so not shallow). 
I have also tried with a virtual python environment running python 3.9 but get the same issue.

**Environment**
- Host Machine OS: Mac OS sonoma 14.1.1
- Python version: 3.12

[logs.pdf](https://github.com/espressif/esp-matter/files/14461622/logs.pdf)

**Comment by sayondeep** [comment]
@S-A-M-J Assuming your problem is exactly same as [this](https://www.esp32.com/viewtopic.php?t=38114). Can you please try to install esp-matter in some other directory and not inside the espressif tools ( /Users/<user-name>/.espressif/) directory.

**Comment by S-A-M-J** [comment]
Ok sry apparently it's not exactly the same issue. My install try was not within the espressif toolchain directory but rather in home/esp next to esp-idf

**Comment by S-A-M-J** [comment]
So i installed on a fresh ubuntu system with python 3.8 and that worked. So it must be a mac or python>=3.8 version issue. I assume it is a mac related issue.

**Comment by FHFS** [comment]
I'd recommend the [esp-matter container](https://hub.docker.com/r/espressif/esp-matter/tags)
It saves a lot of time setting things up. You don't even need to init the connectedhomeip repos.
The python virtual env is already set up properly

**Comment by MohammadAbbasi76** [comment]
Hello,

If you encounter issues during installation while using [Matter deployment](https://docs.espressif.com/projects/esp-matter/en/latest/esp32/developing.html) and face Python environment problems, such as

Screenshot from 2024-03-26 12-57-09

or similar issues, ensure you have Python 3.11 installed as a first step. If the problem persists, proceed to the directory /.espressif/esp-matter/connectedhomeip/connectedhomeip/scripts and execute bootstrap.sh. This step might resolve the issue.

**Comment by shubhamdp** [comment]
@MohammadAbbasi76 I think you may have missed sourcing `$IDF_PATH/export.sh`, can you please run `source $IDF_PATH/export.sh` and try running the command again?

**Comment by hamslabs** [comment]
I'd seeing the same issue as @S-A-M-J on my M2 MacBook Air. The output I see is virtually the same. Is there any more verbose trace that would provide more info for you folks?

Also @FHFS, that docker image doesn't work either. I see this error dialog:

![image](https://github.com/espressif/esp-matter/assets/599838/50e88051-e76d-4912-8095-98a7deb9aed6)

**Comment by MohammadAbbasi76** [comment]
[shubhamdp](https://github.com/shubhamdp)I use Ubuntu 23.10 and  I make it fix with  virtual environment using venv or virtualenv in python ,
![Screenshot from 2024-05-07 10-38-24](https://github.com/espressif/esp-matter/assets/104908459/3719ee89-fbe9-4f88-b69a-472d92187959)

```
sudo apt install python3-venv
python3 -m venv .venv
```

this [link](https://askubuntu.com/questions/1465218/pip-error-on-ubuntu-externally-managed-environment-%C3%97-this-environment-is-extern) that you can use it .
first step in make virtual environment for python after that run `./install.sh `

**Comment by m0rt4l1n** [comment]
Try changing 
```json
"platforms": ["mac-amd64", "windows-amd64"],
"tags": ["version:2@3.9.5.chromium.19"]
```
To 
```json
"platforms": ["mac-amd64", "windows-amd64","mac-arm64"],
"tags": ["version:2@3.11.9.chromium.35"]
```
in file `connectedhomeip/connectedhomeip/scripts/setup/python.json`

(I am on branch `releases/v1.2` and this worked for me on M2 Pro)
(I checked [here](https://chrome-infra-packages.appspot.com/p/infra/3pp/tools/cpython3/mac-arm64/) for available tags with `mac-arm64` versions)

**Comment by hamslabs** [comment]
I tried the workaround that @m0rt4l1n suggested and it worked on my M2 MacBook Air using the main branch. Thanks much.

**Comment by shubhamdp** [comment]
Python3.13 version related fixes(a6c9e5b9c72e3a89a700d7650344f92389c71f58 and  e09e20b693d1921f997f373055f92fb04e395fb1) are merged in main, so this can be closed!
